webapp/src/Service/SubmissionService.php

```diff
@@ -512,11 +512,16 @@ public function submitSolution(
 
         $this->logger->info('Submission input verified');
 
-        // First look up any expected results in file, so as to minimize the
+        // First look up any expected results in all submission files, so as to minimize the
         // SQL transaction time below.
         if ($this->dj->checkrole('jury')) {
-            $results = self::getExpectedResults(file_get_contents($files[0]->getRealPath()),
-                $this->config->get('results_remap'));
+            foreach ($files as $rank => $file) {
+                $results = self::getExpectedResults(file_get_contents($file->getRealPath()),
+                    $this->config->get('results_remap'));
+                if ($results !== null){
+                    break;
+                }
```

Comment:
I assume this is as you expect that only one file has the "EXPECTED_RESULT" which I think should make sense, but as this is for jury solutions (which are not run often and are critical) must this not be really strict and fail if such a mistake is made? Or merge all the expected results?

Although I suspect this will probably (almost) never happen in for real as those are generated via other tools and multifile does not happen often I think? But for this case being strict might be preferred.

Reply:
Yeah this could be more strict. In that case we should also check there's only a single match per file inside `getExpectedResult`.

Also, I don't know how exactly this interacts with the problem archive format import, but in BAPCtools I'm now requiring that a submission in e.g. the wrong answer directory must have the corresponding tag in the expected results list. (It would be weird to only have the time limit exceeded tag in this case.)